app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java

```diff
@@ -285,16 +288,22 @@ private void changeHomeBackground() {
     }
   }
 
-  private boolean isLocationServicesEnabled(Context context) {
+  private boolean isLocationServicesEnabled() {
     LocationManager locationManager =
-        (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+        (LocationManager) getContext().getSystemService(Context.LOCATION_SERVICE);
     boolean gpsEnabled;
     boolean networkEnabled;
 
     gpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
     networkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
 
-    return gpsEnabled || networkEnabled;
+    if (!gpsEnabled && !networkEnabled) {
```

Comment:
if (!gpsEnabled || !networkEnabled) because the dialog should be shown if either gps or network is not enabled right? Something is not right here.

Reply:
I think this is fine